### PR TITLE
Add assertEventually overload with retryDuration

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Review by raunaqmorarka — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java [hunk 1]
```diff
@@ -58,6 +59,11 @@ public static void assertEventually(Runnable assertion)
     }
 
     public static void assertEventually(Duration timeout, Runnable assertion)
+    {
+        assertEventually(timeout, new Duration(50, MILLISECONDS), assertion);
+    }
+
+    public static void assertEventually(Duration timeout, Duration retryDuration, Runnable assertion)
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java:66]: retryDuration -> retryFrequency or waitBeforeRetry

File: core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java [hunk 2]
```diff
@@ -71,7 +77,7 @@ public static void assertEventually(Duration timeout, Runnable assertion)
                 }
             }
             try {
-                Thread.sleep(50);
+                Thread.sleep(retryDuration.toMillis());
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java:80]: i wonder what are the cases when 50ms is a wrong value.
Maybe we can derive a better waitBeforeRetry value based on timeout?
Eg. for timeout in minutes, sleeping second or few seconds sounds more reasonable

Comment (findepi) [reply core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java:80]: for example, the sleep could be 1/1000 of timeout eliminating the need for a parameter

**Review by findepi — APPROVED** [review]
% arg naming change 

the suggestion to go without explicit param is a nice to have

File: core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java (hunk 1 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java:66]: retryFrequency is nice

File: core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java (hunk 2 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java:80]: I like that. Not sure about the `1/1000` though. What are timeouts typically?

**Review by losipiuk — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java (hunk 2 above)
Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java:80]: > Maybe we can derive a better waitBeforeRetry value based on timeout?

That can be done independently of this PR

**Pull request merged**